src/logic/treeBuilder/treeBuilder.ts

```diff
@@ -0,0 +1,24 @@
+import * as _ from "lodash";
+import { TreeNode } from "./treeNode";
+
+export interface TreeBuildParams {
+  branchFactor: number; // number of childrens for each node
+  treeDepth: number; // number of levels except root
+}
+
+export class TreeBuilder {
+  static build({ branchFactor, treeDepth }: TreeBuildParams): TreeNode {
```

Comment:
Dość prosta metoda do budowania drzewa. 
Dostaje dwa argumenty:
`branchFactor` oraz `treeDepth` i na podstawie tego tworzy drzewo, które ma `treeDepth` poziomów i w którym każde dziecko ma `branchFactor` dzieci

Reply:
Myślę, że można by było to ulepszyć w taki sposób, że dodać do niego jakiś element losowości:
np. liczba dzieci - liczba z przedziału `1 .. branchFactor`